tests/zfs-tests/tests/functional/dbufs/dbufstats_002_pos.ksh

```diff
@@ -0,0 +1,87 @@
+#!/bin/ksh -p
+#
+# CDDL HEADER START
+#
+# The contents of this file are subject to the terms of the
+# Common Development and Distribution License (the "License").
+# You may not use this file except in compliance with the License.
+#
+# You can obtain a copy of the license at usr/src/OPENSOLARIS.LICENSE
+# or http://www.opensolaris.org/os/licensing.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+#
+# When distributing Covered Code, include this CDDL HEADER in each
+# file and include the License file at usr/src/OPENSOLARIS.LICENSE.
+# If applicable, add the following below this CDDL HEADER, with the
+# fields enclosed by brackets "[]" replaced with your own identifying
+# information: Portions Copyright [yyyy] [name of copyright owner]
+#
+# CDDL HEADER END
+#
+
+#
+# Copyright (c) 2016, Lawrence Livermore National Security, LLC.
+#
+
+. $STF_SUITE/tests/functional/dbufs/dbufs.kshlib
+. $STF_SUITE/include/libtest.shlib
+. $STF_SUITE/include/math.shlib
+
+#
+# DESCRIPTION:
+# Ensure that dbufs move from mru to mfu as expected.
+#
+# STRATEGY:
+# 1. Set dbuf cache size to a small size (10M for this test)
+# 2. Generate a file with random data (small enough to fit in cache)
+# 3. zpool sync to remove dbufs from anon list in ARC
+# 4. Obtain the object ID using linux stat command
+# 5. Ensure that all dbufs are on the mru list in the ARC
+# 6. Generate another random file large enough to flush dbuf cache
+# 7. cat the first generated file
+# 8. Ensure that at least some dbufs moved to the mfu list in the ARC
+#
+# NOTES:
+# Currently, when dbufs are marked as DB_CACHED in the dbuf cache, the
+# ARC header for the dbuf is not updated to reflect the "hit". Therefore,
+# the only way to cause a dbuf to transition from mru to mfu is to ensure
+# that it is in the ARC, but not in the dbuf cache.
+#
+
+DBUFS_FILE="/var/tmp/dbufs.out.$$"
+
+function cleanup
+{
+	log_must rm -f $TESTDIR/file $TESTDIR/file2 $DBUFS_FILE
+}
+
+verify_runnable "both"
+
+log_assert "dbufs move from mru to mfu list"
+
+log_onexit cleanup
+
+log_must dd if=/dev/urandom bs=1M count=1 of="$TESTDIR/file"
```

Comment:
You could make this more portable by using ZFS's `file_write` with the psudorandom option:
```file_write -o create -f "$TESTDIR/file" -b 1048576 -c 1 -d R```